src/thinblock.cpp

```diff
@@ -163,7 +163,7 @@ bool CThinBlock::process(CNode *pfrom, int nSizeThinBlock)
 
     {
         LOCK(cs_orphancache);
-        LOCK2(mempool.cs, cs_xval);
+        LOCK(cs_xval);
         int missingCount = 0;
```

Comment:
I believe that thanking out the mempool lock here breaks the lockorder....  mempool.cs has to be taken before cs_xval

Reply:
It shouldn't because you can't get into a deadlock situation just by removing a lock.  You need to reverse the lock order.  Note that if this code locked mempool.cs again below cs_xval then removing the lock here would change the locking order.  So this might be the source of your confusion.  

Just to be sure that the checker does not incorrectly flag this, I ran the following code under a debug build and it works:
    {
         LOCK(cs_orphancache);
         READLOCK(mempool.cs);
         LOCK(cs_xval);
         printf("l1\n");
    }
    {
         LOCK(cs_orphancache);
         LOCK(cs_xval);
         printf("l2\n");
    }

Reply:
There used to be a lock order constraint on these three locks, however, after looking through your code again, i see that you've removed this restriction (which is now possible) because of your read locks.  This is great!!!  I don't have any other issues with this PR...will approve now.